lib/sneakers/queue.rb

```diff
@@ -59,6 +59,8 @@ def unsubscribe
     # XXX can we cancel bunny and channel too?
     @consumer.cancel if @consumer
     @consumer = nil
+  rescue Bunny::ConnectionClosedError
+    sleep @bunny.instance_variable_get(:@network_recovery_interval)
```

Comment:
This is not right. If you'd like `Bunny::Session` to expose its effective network recovery interval, please file an issue and we will add a getter.